### ISSUE TITLE
DebuggerTypeProxy RawView must come first

### ISSUE THREAD

**Issue by MkazemAkhgary** [dotnet/roslyn#22821]
when a collection has 1000 items, ill see 1000 items first and I have to scroll down all over (and wait for a lot of time) to see RawView at end.

Why is it like that? it should be at first so some one doesn't want to see 1000 items in collection and just want to access and view internal fields can do that without trouble.

![image 3](https://user-images.githubusercontent.com/15255162/31938624-e81799b4-b8c4-11e7-800a-03040640d9e4.png)

I want to access this without scrolling down!

![image 4](https://user-images.githubusercontent.com/15255162/31938662-06f551fa-b8c5-11e7-91c1-126d0db293bc.png)

**Comment by benaadams** [comment]
Something like switching [DebuggerTypeProxyExpansion.cs](https://github.com/dotnet/roslyn/blob/fa7f6c9f86f537a93142a96fc688a9147b122491/src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/DebuggerTypeProxyExpansion.cs#L175-L183) from
```csharp
if (_proxyItem != null)
{
    _proxyItem.Expansion.GetRows(resultProvider, rows, inspectionContext, _proxyItem.ToDataItem(), _proxyItem.Value, startIndex, count, visitAll, ref index);
}

if (InRange(startIndex, count, index))
{
    rows.Add(this.CreateRawViewRow(resultProvider, inspectionContext));
}
```
To
```csharp
if (InRange(startIndex, count, 0))
{
    rows.Add(this.CreateRawViewRow(resultProvider, inspectionContext));
    count--;
}

if (_proxyItem != null)
{
    _proxyItem.Expansion.GetRows(resultProvider, rows, inspectionContext, _proxyItem.ToDataItem(), _proxyItem.Value, startIndex, count, visitAll, ref index);
}
```
?

**Comment by MkazemAkhgary** [comment]
@benaadams yes please.

**Comment by jcouv** [comment]
@jinujoseph Could you help route/triage this?

**Comment by tmat** [comment]
I don't think we want to move Raw View to the front. Raw View is not not what most users want to see. 

That said, I think we can improve visualization of arrays in VS. For example, VS Mac displays them using a tree view, where you can dig into parts of the array by groups, e.g.:
```
+ [0-999]
   + [0-99]
   + [100-199]
       ...
   + [900-999]
+ [1000-1999]
+ [2000-2541]
Raw View
```

On every level there is at most 10 entries. Getting to the end is easy.

@KayAnd

**Comment by MkazemAkhgary** [comment]
@tmat that will work too. It would be also good if you wanted to jump to specified index. When i want to debug a parser i want to see content of arrays, they are not small. That forces me to write separate code to see content.

A textbox that is always visible would help.

**Comment by benaadams** [comment]
@tmat chrome takes a similar approach in is dev console - its a very useful representation

**Comment by ivanbasov** [comment]
This is a feature request. It does not meet the bar for 15.6. Punting
